crates/local-cluster-runner/src/node/mod.rs

```diff
@@ -0,0 +1,740 @@
+        .args(&args);
+
+        let mut child = cmd.spawn().map_err(NodeStartError::SpawnError)?;
+        let pid = child.id().expect("child to have a pid");
+        info!(
+            "Started node {} in {} (pid {pid})",
+            base_config.node_name(),
+            node_base_dir.display(),
+        );
+        let stdout = child.stdout.take().expect("child to have a stdout pipe");
+        let stderr = child.stderr.take().expect("child to have a stderr pipe");
+
+        let stdout_reader =
+            stream::try_unfold(BufReader::new(stdout).lines(), |mut lines| async move {
+                match lines.next_line().await {
+                    Ok(Some(line)) => Ok(Some((line, lines))),
+                    Ok(None) => Ok(None),
+                    Err(err) => Err(err),
+                }
+            });
+        let stderr_reader =
+            stream::try_unfold(BufReader::new(stderr).lines(), |mut lines| async move {
+                match lines.next_line().await {
+                    Ok(Some(line)) => Ok(Some((line, lines))),
+                    Ok(None) => Ok(None),
+                    Err(err) => Err(err),
+                }
+            });
+
+        let lines = futures::stream::select(stdout_reader, stderr_reader);
+        let searcher = Searcher::new();
+
+        let lines_fut = {
+            let searcher = searcher.clone();
+            async move {
+                let searcher = &searcher;
+                let mut node_log_file = lines
+                    .try_fold(node_log_file, |mut node_log_file, line| async move {
+                        node_log_file.write_all(line.as_bytes()).await?;
+                        node_log_file.write_u8(b'\n').await?;
+                        searcher.matches(line.as_str()).await;
+                        Ok(node_log_file)
+                    })
+                    .await?;
+                node_log_file.flush().await?;
+                searcher.close();
+                io::Result::Ok(())
+            }
+        };
+
+        let node_name = base_config.node_name().to_owned();
+        let child_handle = tokio::spawn(async move {
+            let (status, _) = tokio::join!(child.wait(), lines_fut);
+
+            match status {
+                Ok(status) => {
+                    info!("Node {} exited with {status}", node_name);
+                }
+                Err(ref err) => {
+                    error!(
+                        "Node {} exit status could not be determined: {err}",
+                        node_name
+                    );
+                }
+            }
+
+            status
+        });
+
+        Ok(StartedNode {
+            status: StartedNodeStatus::Running {
+                child_handle,
+                searcher,
+                pid,
+            },
+            config: base_config,
+        })
+    }
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[cfg_attr(feature = "schemars", derive(schemars::JsonSchema))]
```

Comment:
I wonder if we need JsonSchema in any of this. Do you have a use-case in mind?

Reply:
no, we dont i agree